File: problems/0007_reverse-integer/Solution.scala
object Solution {
    def reverse(x: Int): Int = {
        var n = x
        var result = 0L
        while (n != 0) {
            result = result * 10 + (n % 10)
            n /= 10
        }
        if (result < Int.MinValue || Int.MaxValue < result)
            0
        else
            result.toInt
    }
}

File: problems/0007_reverse-integer/Solution.cs
public class Solution {
    public int Reverse(int x) {
        long result = 0;
        for (; x != 0; x /= 10)
            result = result * 10 + (x % 10);
        return (result < Int32.MinValue || Int32.MaxValue < result ?
               0 : (int) result);
    }
}
